src/formatDate.js

```diff
@@ -50,7 +50,35 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const delimiter = fromFormat[fromFormat.length - 1];
+  const dateArr = date.split(delimiter);
+  const result = [];
+  const shortYear = 'YY';
+  const dateObj = {
+    [fromFormat[0]]: dateArr[0],
+    [fromFormat[1]]: dateArr[1],
+    [fromFormat[2]]: dateArr[2],
+  };
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    if (toFormat[i].includes(shortYear)) {
+      const year = dateObj.hasOwnProperty(shortYear)
+        ? dateObj[shortYear]
+        : dateObj['YYYY'];
+
+      if (!dateObj.hasOwnProperty(toFormat[i])) {
+        result.push(year < 30
+          ? `20${year}`
+          : `${year.slice(2)}` || `19${year}`);
+      } else {
+        result.push(year);
+      }
```

Comment:
one more solution to avoid this `if else`

```suggestion
     const formattedYear = year < 30
       ? `20${year}`
       : `${year.slice(2)}` || `19${year}`;

     const needYear = !dateObj.hasOwnProperty(toFormat[i]) ? formattedYear : year

     result.push(needYear);
```